### DISCUSSION TITLE
Function that takes callable with generics

### DISCUSSION THREAD

**Discussion by sylee957** [microsoft/pyright#4919] [Q&A] [answered]
I can't match argument `tuple[X] -> Y` with `tuple[int] -> int` function in the following example.

```python3
from collections.abc import Callable
from typing import TypeVar

X = TypeVar("X")
Y = TypeVar("Y")


def unwrap(func: Callable[[tuple[X]], Y]) -> Callable[[X], Y]:
    def wrapped(x: X) -> Y:
        return func((x,))

    return wrapped


def f(x: tuple[int]) -> int:
    return x[0] + 1
```

`unwrap(f)` does not pass the type check.

**Comment by erictraut** [comment, marked as answer]
Yes, this is a bug. Thanks for reporting.

**Reply by erictraut** [reply]
This is addressed in pyright 1.1.303, which I just published. The change will be included in a future version of pylance.
